pygfx/utils/transform.py

```diff
@@ -150,9 +150,19 @@ def matrix(self):
     def _inverse_matrix(self) -> np.ndarray:
         return np.linalg.inv(self.matrix)
 
+    @property
+    def scaling_signs(self):
+        return self._scaling_signs
+
     @cached
     def _decomposed(self) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
-        return la.mat_decompose(self.matrix, scaling_signs=self._scaling_signs)
+        try:
+            return la.mat_decompose(self.matrix, scaling_signs=self.scaling_signs)
+        except ValueError:
+            # the matrix has been set manually
+            # and so there is no scaling component to preserve
+            # any decomposed scaling is acceptable
+            return la.mat_decompose(self.matrix)
```

Comment:
How does the `ValueError`  come about? Can/should we be more precise in detecting when the matrix has been manually set?

Reply:
The ValueError is raised when the provided signs don't align with the determinant of the matrix (even/odd number of signs).

All codepaths use the matrix setter so that makes detecting manual matrix setting inconvenient.

I figured, if we're going to set an arbitrary sign anyway, it doesn't matter if we use self._scaling_signs or not! So we can just try it and fall back to not using it.